lib/xapi-stdext-date/date.ml

```diff
@@ -12,18 +12,29 @@
  * GNU Lesser General Public License for more details.
  *)
 
-type iso8601 = string
+(** this dash type is used to store information about whether
+    the input string contained dashes or not (e.g. it is false when
+    `of_float` is called)
+  * we remember the dashes only where possible:
+     - `"yyyy-mm-ddTxx" |> of_string |> to_string = "yyyy-mm-ddTxx"
+  *  - `"yyyy-mm-ddTxx" |> of_string |> to_float |> of_float |> to_string = yyyymmddTxx`
+  *)
+type dash = bool
```

Comment:
Why is this needed, wouldn't changing the of_string function be enough to accept the format with dashes?

Reply:
it's to ensure that `of_string . to_string` is the identity function

i imagine that it isn't necessary, but if somebody saved one format to the database, for example, they might expect the same format to come out again

Reply:
Can that property can be kept on a serialization + deserialization? If not I wouldn't worry about it, that's the main use-case of the library.